Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -435,7 +435,16 @@ private func createResolvedPackages(
                 let explicitIdsOrNames = Set(explicitProducts.lazy.map({ lookupByProductIDs ? $0.identity : $0.name }))
                 return dependency.products.filter({ lookupByProductIDs ? explicitIdsOrNames.contains($0.product.identity) : explicitIdsOrNames.contains($0.product.name) })
             })
-        let productDependencyMap = lookupByProductIDs ? productDependencies.spm_createDictionary({ ($0.product.identity, $0) }) : productDependencies.spm_createDictionary({ ($0.product.name, $0) })
+
+        let productDependencyMap: [String: ResolvedProductBuilder]
+        if lookupByProductIDs {
+            productDependencyMap = productDependencies.spm_createDictionary({ ($0.product.identity, $0) })
+        } else {
+            productDependencyMap = try Dictionary(productDependencies.map({ ($0.product.name, $0) }), uniquingKeysWith: { lhs, rhs in
+                let duplicates = productDependencies.filter { $0.product.name == lhs.product.name }
+                throw PackageGraphError.duplicateProduct(product: lhs.product.name, packages: duplicates.map { $0.packageBuilder.package.identity.description })
```

Comment:
formatting nit: this exceeds 120 characters limit set in SwiftFormat config

Reply:
🫠

Reply:
I'm totally fine with keeping it as is, seems like we agreed that reformatting everything at once in a future PR is the way to go 🫣

Reply:
Sorry, wasn't a reaction to you but to the arbitrary SwiftFormat config. Thanks for pointing it out!